src/Sfx/App/Scripts/Models/DataModels/Cluster.ts

```diff
@@ -40,6 +65,61 @@ module Sfx {
+            Certificate expiration: thumbprint = 35d8f6bb4c52bd3f40a327a3094a9ee9692679ce, expiration = 2020-03-13 22:23:40.000
+            , remaining lifetime is 213:8:17:08.174, please refresh ahead of time to avoid catastrophic failure.
+            Warning threshold Security/CertificateExpirySafetyMargin is configured at 289:8:26:40.000, if needed, you can
+            adjust it to fit your refresh process.
+
+            */
+
+            const thumbprintSearchText = "thumbprint = ";
+            const thumbprintIndex = healthEvent.raw.Description.indexOf(thumbprintSearchText);
+            const thumbprint =  healthEvent.raw.Description.substr(thumbprintIndex + thumbprintSearchText.length).split(",")[0];
+
+            const expirationSearchText = "expiration = ";
+            const expirationIndex = healthEvent.raw.Description.indexOf("expiration = ");
+            const expiration = healthEvent.raw.Description.substring(expirationIndex + expirationSearchText.length).split(",")[0];
+
+            this.data.warnings.addOrUpdateNotification({
+                message: `A cluster certificate is set to expire soon. Replace it as soon as possible to avoid catastrophic failure. <br> Thumbprint : ${thumbprint}    Expiration: ${expiration}`,
+                level: StatusWarningLevel.Error,
+                priority: 5,
+                id: BannerWarningID.ExpiringClusterCert,
+                link: "https://aka.ms/sfrenewclustercert/",
+                linkText: "Read here for more guidance"
+            });
+            ClusterHealth.certExpirationChecked = true;
+        }
+
+        private containsCertExpiringHealthEvent(unhealthyEvaluations: HealthEvent[]): HealthEvent[] {
+            return unhealthyEvaluations.filter(event => event.raw.Description.indexOf("Certificate expiration") === 0 &&
+                                             event.raw.Property === CertExpiraryHealthEventProperty.Cluster &&
+                                             (event.raw.HealthState === HealthStateConstants.Warning || event.raw.HealthState === HealthStateConstants.Error));
+        }
+
+        private checkNodesContinually(index: number, nodes: Node[]) {
+            if (index < nodes.length) {
+                const node = nodes[index];
+                if (node.healthState.text === HealthStateConstants.Error || node.healthState.text === HealthStateConstants.Warning) {
+                    node.health.ensureInitialized().then( () => {
+                        const certExpiringEvents = this.containsCertExpiringHealthEvent(node.health.healthEvents);
+                        if (certExpiringEvents.length === 0) {
+                            this.checkNodesContinually(index + 1, nodes);
+                        }else {
+                            this.setMessage(certExpiringEvents[0]);
+                        }
+                    });
+                }else {
+                    this.checkNodesContinually(index + 1, nodes);
```

Comment:
What interval does this run at?

Reply:
so it'll recursively check nodes one at a time so it runs at an interval of a round trip.

Reply:
Do we want to add some time between round trips so that there's a minimum round trip time?

Reply:
Misread this portion, sorry :)